modules/spechla/1.0/spechla.smk

```diff
@@ -0,0 +1,310 @@
+#!/usr/bin/env snakemake
+
+
+##### ATTRIBUTION #####
+
+
+# Original Author:  Jasper Wong
+# Module Author:    Jasper Wong
+# Contributors:     N/A
+
+
+##### SETUP #####
+
+# Import package with useful functions for developing analysis modules
+import oncopipe as op
+
+# Check that the oncopipe dependency is up-to-date. Add all the following lines to any module that uses new features in oncopipe
+min_oncopipe_version="1.0.11"
+import pkg_resources
+try:
+    from packaging import version
+except ModuleNotFoundError:
+    sys.exit("The packaging module dependency is missing. Please install it ('pip install packaging') and ensure you are using the most up-to-date oncopipe version")
+
+# To avoid this we need to add the "packaging" module as a dependency for LCR-modules or oncopipe
+
+current_version = pkg_resources.get_distribution("oncopipe").version
+if version.parse(current_version) < version.parse(min_oncopipe_version):
+    print('\x1b[0;31;40m' + f'ERROR: oncopipe version installed: {current_version}' + '\x1b[0m')
+    print('\x1b[0;31;40m' + f"ERROR: This module requires oncopipe version >= {min_oncopipe_version}. Please update oncopipe in your environment" + '\x1b[0m')
+    sys.exit("Instructions for updating to the current version of oncopipe are available at https://lcr-modules.readthedocs.io/en/latest/ (use option 2)")
+
+# End of dependency checking section
+
+# Setup module and store module-specific configuration in `CFG`
+# `CFG` is a shortcut to `config["lcr-modules"]["spechla"]`
+CFG = op.setup_module(
+    name = "spechla",
+    version = "1.0",
+    subdirectories = ["inputs", "hla_reads", "spechla", "loh", "outputs"],
+)
+
+# Define rules to be run locally when using a compute cluster
+localrules:
+    _spechla_input_bam,
+    _spechla_setup,
+    _spechla_index,
+    _spechla_output_txt,
+    _spechla_all,
+
+
+##### RULES #####
+
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _spechla_input_bam:
+    input:
+        bam = CFG["inputs"]["sample_bam"],
+        bai = CFG["inputs"]["sample_bai"]
+    output:
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam",
+        bai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bai",
+        crai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.crai"
+    group: 
+        "symlink"
+    run:
+        op.absolute_symlink(input.bam, output.bam)
+        op.absolute_symlink(input.bai, output.bai)
+        op.absolute_symlink(input.bai, output.crai)
+
+
+# Example variant calling rule (multi-threaded; must be run on compute server/cluster)
+rule _spechla_setup:
+    output:
+        txt = touch(CFG["dirs"]["inputs"] + "SpecHLA/clone.done")
+    log:
+        stdout = CFG["logs"]["inputs"] + "git_clone.stdout.log"
+    params:
+        scriptdir = CFG["dirs"]["inputs"] + "SpecHLA"
+    shell:
+        op.as_one_line("""
```

Comment:
Do you have an example of how to configure this? I'm not too sure how to do a version-control for git pulls here

Reply:
This might be an example
https://github.com/LCR-BCCRC/lcr-modules/blob/3de42f1320ffbe97c2024d93550d2e7433165620/modules/rainstorm/1.0/rainstorm.smk#L57-L69

Reply:
Changed to version-based approach of specHLA set-up